core/src/forest/ForestTrainer.cpp

```diff
@@ -39,11 +39,18 @@ const Forest ForestTrainer::train(Data* data,
   size_t num_samples = data->get_num_rows();
   uint num_trees = options.get_num_trees();
 
-  // Ensure that the sample fraction is not too small.
+  // Ensure that the sample fraction is not too small and honesty fraction is not too extreme.
+    bool honesty = TreeOptions(options.get_tree_options()).get_honesty();
+    double honestyfraction = TreeOptions(options.get_tree_options()).get_honesty_fraction();
```

Comment:
I don't think we need to wrap `options.get_tree_options()` in another constructor call (`TreeOptions(...)`).